### DISCUSSION TITLE
When building can i target only  .net 5 to build not .net 3.1 ?

### DISCUSSION THREAD

**Discussion by giannik** [OrchardCMS/OrchardCore#8716] [Q&A]
Now that Multi targeting frameworks is supported, Im seeing that in debug folder 2 folders are created for each .net 3.1 and .net 5 framework.
Is there a way to instruct visual studio to only build for .net 5 safely, because Im noticing its taking more time to build for both frameworks ?

**Comment by Skrypt** [comment]
You can specify the target framework when you build it.

In VS Code you simply need to add the --framework param in the tasks.json file.

```JSON
    {
      "label": "Web App: Dotnet Build Debug",
      "type": "shell",
      "command": "dotnet",
      "args": [
        "build",
        "${workspaceRoot}/src/OrchardCore.Cms.Web",
        "-c",
        "Debug",
        "--framework",
        "net5.0",
      ],
      "problemMatcher": "$msCompile"
    },
```` 

In Visual Studio : 

![image](https://user-images.githubusercontent.com/3228637/109681059-15985300-7b4b-11eb-9978-3ccb796c4b23.png)

Even if you pick net5.0 here in Visual Studio it will build both. Maybe there is something missing for Visual Studio but normally it should work. Here though what it does is that it uses net5.0 for debugging.

**Reply by jtkech** [reply]
I think there is also a way to force a TFM under VStudio, e.g. through a json file, can't right now but i will check it

**Reply by Skrypt** [reply]
Or just use the command line directly.

```cmd
dotnet build /src/OrchardCore.Cms.Web -c Debug --framework net5.0
```

**Reply by jtkech** [reply]
Yes ;)
